tests/APIv2/RolesTest.php

```diff
@@ -187,6 +187,53 @@ public function testPatchPermissionEndpoint() {
         $this->assertFalse($this->hasPermission('site.manage', 'global', $permissions));
         $this->assertFalse($this->hasPermission('comments.add', 'category', $permissions, 1));
     }
+    /**
+     * Test updating permissions with PATCH /roles/:id/permissions
+     */
+    public function testPatchPermissionOverWrite() {
+        $role = $this->getPermissionsRole([[
+            'type' => 'category',
+            'id' =>  1,
+            'permissions' => [
+                'discussions.view' => true,
+                'discussions.add' => true,
+                'comments.add' => true
+            ]
+           ]]);
+
+        $role2 = $this->getPermissionsRole([[
+            'type' => 'category',
+            'id' => 1,
+            'permissions' => [
+                'discussions.view' => true,
+                'discussions.add' => true,
+                'comments.add' => true
+            ]
+        ]]);
+
+        $this->api()->patch(
+            "{$this->baseUrl}/{$role['roleID']}/permissions",
+            [
+                [
+                    'type' => 'category',
+                    'id' => 1,
+                    'permissions' => [
+                        'discussions.add' => true,
+                        'comments.add' => false
+                    ]
+                ]
+            ]
+        );
+
+        $permissions1 = $this->getPermissions($role['roleID']);
+        $permissions2 = $this->getPermissions($role2['roleID']);
+
+        $this->assertTrue($this->hasPermission('discussions.add', 'category', $permissions1, 1));
+        $this->assertFalse($this->hasPermission('comments.add', 'category', $permissions1, 1));
```

Comment:
Why do we still expect 1 for `comments.add` ?
We just patched it few lines before to false?

Reply:
assertFalse verfies the condition is false from my understanding.

https://phpunit.readthedocs.io/en/7.4/assertions.html#assertfalse

Reply:
:-) My bad sorry! :-)